src/ir/operation.h

```diff
@@ -0,0 +1,54 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#ifndef SRC_IR_OPERATION_H_
+#define SRC_IR_OPERATION_H_
+
+#include <vector>
+
+#include "absl/container/flat_hash_map.h"
+#include "src/ir/operator.h"
+#include "src/ir/types/type.h"
+#include "src/ir/value.h"
+
+namespace raksha::ir {
+
+// An Operation represents a unit of execution.
+class Operation {
+ public:
+  Operation(const Block* parent, const Operator& op, NamedValueListMap inputs)
```

Comment:
Right now, eliding a bit of detail, the structure is:
- `Block`s have a list of `Operation`s
- `Operation`s have an `Operator`
- `Operator`s have a block.

But why do we need both operations and operators? Can we just give `Operation` a name and a pointer to a Block for its implementation and then drop `Operator`s?